evmrpc/tx.go

```diff
@@ -270,19 +269,37 @@ func getEthTxForTxBz(tx tmtypes.Tx, decoder sdk.TxDecoder) *ethtypes.Transaction
 	return ethtx
 }
 
-func encodeReceipt(receipt *types.Receipt, blockHash bytes.HexBytes) (map[string]interface{}, error) {
+func encodeReceipt(receipt *types.Receipt, decoder sdk.TxDecoder, block *coretypes.ResultBlock) (map[string]interface{}, error) {
+	blockHash := block.BlockID.Hash
 	bh := common.HexToHash(blockHash.String())
 	logs := keeper.GetLogsForTx(receipt)
 	for _, log := range logs {
 		log.BlockHash = bh
 	}
+	// convert tx index including cosmos txs to tx index excluding cosmos txs
+	evmTxIndex := 0
+	foundTx := false
+	for _, tx := range block.Block.Txs {
+		etx := getEthTxForTxBz(tx, decoder)
+		if etx == nil { // cosmos tx, skip
+			continue
+		}
+		if etx.Hash() == common.HexToHash(receipt.TxHashHex) {
```

Comment:
would be cheaper to just check `if i == receipt.TransactionIndex` where `for i, tx := range ...`